peregrine/blueprints/datasets.py

```diff
@@ -60,3 +61,33 @@ def get_datasets():
         node = match.group(2)
         result[projects[index]][node] = value
     return flask.jsonify(result)
+
+
+@blueprint.route("/projects", methods=["GET"])
+def get_projects():
+    """
+    Get all projects high level information, if a deployment is configured
+    to set PUBLIC_DATASETS to True, this endpoint will be open to
+    anonymous users
+    """
+    if os.environ.get("PUBLIC_DATASETS", "false").lower() == "true":
+        set_read_access_projects_for_public_endpoint()
+    else:
+        set_read_access_projects()
+    projects = flask.g.read_access_projects
+    if not projects:
+        return flask.jsonify({})
+    # construct a query that get counts for all projects
+    # because graphql can't add structure to group by projects,
+    # we labeled the count by project index and later parse it
+    # with regex to add structure to response
+    query = "{project { name code dbgap_accession_number "
+    for field in ['description', 'image_url']:
+        if dictionary.schema['project']['properties'].get(field):
+            query += field + ' '
+
+    query += '}}'
+    data, errors = graphql.execute_query(query, variables={})
+    if errors:
+        return flask.jsonify({"data": data, "errors": errors}), 400
+    return flask.jsonify({"projects": data["project"]})
```

Comment:
Lets just return `data["project"]` so this matches the swagger?

Reply:
it's more best practice to return object instead of array for json response. I will update in other places